packages/sdk-ts/src/core/modules/exchange/msgs/MsgBatchCancelBinaryOptionsOrders.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { MsgBatchCancelDerivativeOrders as BaseMsgBatchCancelDerivativeOrders } from '@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb'
+import MsgBatchCancelDerivativeOrders from './MsgBatchCancelDerivativeOrders'
+import { mockFactory } from '@injectivelabs/test-utils'
+import snakecaseKeys from 'snakecase-keys'
+
+const params: MsgBatchCancelDerivativeOrders['params'] = {
+  injectiveAddress: mockFactory.injectiveAddress,
+  orders: [
+    {
+      marketId: mockFactory.injUsdtDerivativeMarket.marketId,
+      orderHash: mockFactory.orderHash,
+      subaccountId: mockFactory.subaccountId,
+    },
+    {
+      marketId: mockFactory.injUsdtDerivativeMarket.marketId,
+      orderHash: mockFactory.orderHash2,
+      subaccountId: mockFactory.subaccountId,
+    },
+  ],
+}
+
+const protoType = '/injective.exchange.v1beta1.MsgBatchCancelDerivativeOrders'
+const protoTypeShort = 'exchange/MsgBatchCancelDerivativeOrders'
+
+const ordersWithOrderMask = params.orders.map((order) => ({
+  ...order,
+  orderMask: 1,
+}))
+
+const protoParamsBefore = {
+  sender: params.injectiveAddress,
+  dataList: ordersWithOrderMask,
+}
+
+const protoParamsAfter = {
```

Comment:
`protoParamsBefore` => `protoParams`
`protoParamsAfter` => `protoParamsAmino` or `aminoProtoParams` thats up to you

please change these in the rest of the tests as well